meta-resin-common/recipes-support/resin-device-register/resin-device-register/resin-device-register

```diff
@@ -44,25 +46,28 @@ fi
 while true; do
     if curl -sL -w '%{http_code}' $API_ENDPOINT/ping -o /dev/null | egrep "20[0-9]|30[0-9]" >/dev/null
     then
-        read api_key user_id application_id device_type uuid <<<$(jq -r '.apiKey,.userId,.applicationId,.deviceType // "raspberry-pi",.uuid' $CONFIG_PATH)
+        # The `apiKey` default is mainly so that we don't hit issues where we've already registered
+        # and then everything goes out of line due to an empty apiKey
+        read api_key user_id application_id device_type uuid device_api_key registered_at <<<$(jq -r '.apiKey // .deviceApiKey,.userId,.applicationId,.deviceType,.uuid,.deviceApiKey,.registered_at // empty' $CONFIG_PATH)
+
+        if [ -n "${registered_at}" ]; then
+            echo "[INFO] resin-device-register : Device already registered."
+            exit 0
+        fi
 
         status_code=$(curl -s -X POST -w "%{http_code}" \
-                    -o /var/log/register-device-curl.log \
-                    -d "user=$user_id" \
-                    -d "application=$application_id" \
-                    -d "uuid=$uuid"  \
+                    --output $LOG_FILE \
+                    --data-urlencode "user=$user_id" \
+                    --data-urlencode "application=$application_id" \
```

Comment:
You're right, I think this was targeting an older iteration of the api changes, it's fixed now though